src/visualization.jl

```diff
@@ -23,16 +32,105 @@ function Graph(p::AbstractPetriNet)
   stmts = vcat(statenodes, transnodes)
 
   edges = map(1:nt(p)) do k
-    k_name = tname(p, k)
-    vcat(edgify(countmap(map(x->sname(p,x), inputs(p, k))), k_name, false),
-         edgify(countmap(map(x->sname(p, x), outputs(p, k))), k_name, true))
+    vcat(edgify(countmap(Vector{Any}(inputs(p, k))), k, false),
+         edgify(countmap(Vector{Any}(outputs(p, k))), k, true))
   end |> flatten |> collect
 
   stmts = vcat(stmts, edges)
   g = Graphviz.Digraph("G", stmts; graph_attrs=graph_attrs, node_attrs=node_attrs, edge_attrs=edge_attrs)
   return g
 end
 
+function Subgraph(p::AbstractPetriNet; label="cluster", pre="")
+   statenodes = [Node("\"$(pre)s$s\"",
+                      Attributes(:label => "$(sname(p, s))", :shape=>"circle",
+                                 :color=>"#6C9AC3")) for s in 1:ns(p)]
+
+  transnodes = [Node("\"$(pre)t$k\"",
+                     Attributes(:label => "$(tname(p, k))", :shape=>"square",
+                                :color=>"#E28F41")) for k in 1:nt(p)]
+
+  node_attrs  = Attributes(:shape=>"plain", :style=>"filled", :color=>"white")
+  edge_attrs  = Attributes(:splines=>"splines")
+
+  stmts = vcat(statenodes, transnodes)
+
+  edges = map(1:nt(p)) do k
+    vcat(edgify(countmap(Vector{Any}(inputs(p, k))), k, false; pre=pre),
+         edgify(countmap(Vector{Any}(outputs(p, k))), k, true; pre=pre))
+  end |> flatten |> collect
+
+  stmts = vcat(stmts, edges)
+  g = Graphviz.Subgraph(label, stmts; node_attrs=node_attrs, edge_attrs=edge_attrs)
+end
+
+function Graph(m::Multispan)
+  apex = Subgraph(m.apex; label="clusterApex", pre="ap_")
+
+  leg_graphs = map(enumerate(m.legs)) do (i, l)
+    Subgraph(l.codom; label="clusterLeg$i", pre="leg$(i)_")
+  end
+
+
+  graph_attrs = Attributes(:rankdir=>"LR")
+  node_attrs  = Attributes(:shape=>"plain", :style=>"filled", :color=>"white")
+  edge_attrs  = Attributes(:splines=>"splines")
+
+  t_edges = map(enumerate(m.legs)) do (i, l)
+    map(1:nt(m.apex)) do t
+      Edge(["\"ap_t$t\"", "\"leg$(i)_t$(l.components[:T](t))\""],
+      Attributes(:constraint=>"false", :style=>"dotted"))
+    end
+  end
+
+  s_edges = map(enumerate(m.legs)) do (i, l)
+    map(1:ns(m.apex)) do s
+      Edge(["\"ap_s$s\"", "\"leg$(i)_s$(l.components[:S](s))\""],
+      Attributes(:constraint=>"false", :style=>"dotted"))
+    end
+  end
+  stmts = vcat([apex], leg_graphs, t_edges..., s_edges...)
+  g = Graphviz.Digraph("G", stmts; graph_attrs=graph_attrs, node_attrs=node_attrs, edge_attrs=edge_attrs)
+end
+
+function Graph(p::ACSetTransformation)
+  Graph(Multispan(p.dom, [p]))
+end
+
+function Graph(so::Subobject)
```

Comment:
@jpfairbanks @epatters
Any advice on how to include this feature which is dependent on Catlab 0.12.7? I'd like to allow graphs of ACSetTransformations on earlier versions of Catlab, but compiling this file on anything earlier than 0.12.7 throws an error on this method.

Reply:
I think the right answer is to depend on Catlab 0.12.7 in the project.toml and then require people to update their catlab when they update their AlgPetri. Is there any reason not to do that?

Reply:
I hate to admit it, but with Julia 1.7 in beta and that (hopefully) becoming the next LTS, I think supporting 1.0 is becoming too much of a blocker. I think that that is alright.

Reply:
Sounds good. In order to have it compatible with Catlab 0.12.7, I'll restrict the tests to Julia 1.6.